`GatewayResposive` health checks are broken

**Describe the bug**

<!-- A clear and concise description of what the bug is. -->
The main problem is that in [this code](https://github.com/camunda/zeebe/blob/4f8a898f3c63322aa91c5f5768d4e9c1edc6c26c/gateway/src/main/java/io/camunda/zeebe/gateway/impl/probes/health/ResponsiveHealthIndicator.java#L51-L62) we are closing the `ZeebeClient`, so the first check is passed, but the next one will try to use closed `ZeebeClient`. And we will get this response in `/actuator/health`:
```json
{
  "components": {
    ...
    "livenessGatewayResponsive": {
      "details": {
        "derivedFrom": "ResponsiveHealthIndicator",
        "downTime": "PT2M53.443S",
        "lastSeenDelegateHealthStatus": {
          "details": {
            "error": "java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: Channel shutdown invoked",
  ...
```

**To Reproduce**

<!--
Steps to reproduce the behavior

If possible add a minimal reproducer code sample
- when using the Java client: https://github.com/zeebe-io/zeebe-test-template-java

-->
1. Run Zeebe Gateway with `gateway-responsive` health checks (enabled by default)
2. Call `/actuator/health` several times with pauses (5 seconds would be enough)
3. Obtain this error

**Expected behavior**

<!-- A clear and concise description of what you expected to happen. -->
Health checks should not be broken because the `ZeebeClient` is closed.

**Log/Stacktrace**

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->

<details><summary>Full Stacktrace</summary>
 <p>

```
java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: Channel shutdown invoked
        at java.base/java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture.get(Unknown Source)
        at io.camunda.zeebe.gateway.impl.probes.health.ResponsiveHealthIndicator.health(ResponsiveHealthIndicator.java:57)
        at io.camunda.zeebe.util.health.DelayedHealthIndicator.checkHealth(DelayedHealthIndicator.java:68)
        at jdk.internal.reflect.GeneratedMethodAccessor62.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84)
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: Channel shutdown invoked
        at io.grpc.Status.asRuntimeException(Status.java:539)
        at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487)
        at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:563)
        at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:744)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:723)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        ... 3 more

```

</p>
</details>

**Environment:**
- OS: Linux 3.10.0-1160.53.1.el7.x86_64
- Zeebe Version: 8.2.0-alpha4 (I suppose that versions from 8.2.0-alpha3 are affected because of [this commit](https://github.com/camunda/zeebe/commit/f93c8dcb96f69c2633d0d91621d2bbe4ecbec107))
- Configuration: 3 brokers, 3 gateways, 5 partitions, Elasticsearch exporter.

--- a/gateway/src/main/java/io/camunda/zeebe/gateway/impl/probes/health/ResponsiveHealthIndicator.java
+++ b/gateway/src/main/java/io/camunda/zeebe/gateway/impl/probes/health/ResponsiveHealthIndicator.java
@@ -48,17 +48,16 @@ public class ResponsiveHealthIndicator implements HealthIndicator {
   @Override
   public Health health() {
     Builder resultBuilder;
-    try (final var zeebeClient = supplyZeebeClient()) {
 
-      if (zeebeClient == null) {
-        resultBuilder = Health.unknown();
-      } else {
-        try {
-          zeebeClient.newTopologyRequest().send().get();
-          resultBuilder = Health.up();
-        } catch (final Throwable t) {
-          resultBuilder = Health.down().withException(t);
-        }
+    initZeebeClient();
+    if (zeebeClient == null) {
+      resultBuilder = Health.unknown();
+    } else {
+      try {
+        zeebeClient.newTopologyRequest().send().get();
+        resultBuilder = Health.up();
+      } catch (final Throwable t) {
+        resultBuilder = Health.down().withException(t);
       }
     }
 
@@ -72,12 +71,23 @@ public class ResponsiveHealthIndicator implements HealthIndicator {
         .build();
   }
 
-  ZeebeClient supplyZeebeClient() {
+  void initZeebeClient() {
     if (zeebeClient == null && gatewayCfg.isInitialized()) {
       zeebeClient = createZeebeClient(gatewayCfg, healthZeebeClientProperties);
     }
+  }
 
+  public ZeebeClient getZeebeClient() {
     return zeebeClient;
+  }
+
+  /**
+   * Only for test purposes
+   *
+   * @param zeebeClient new ZeebeClient that should be set
+   */
+  void setZeebeClient(final ZeebeClient zeebeClient) {
+    this.zeebeClient = zeebeClient;
   }
 
   static ZeebeClient createZeebeClient(

--- a/gateway/src/test/java/io/camunda/zeebe/gateway/impl/probes/health/ResponsiveHealthIndicatorTest.java
+++ b/gateway/src/test/java/io/camunda/zeebe/gateway/impl/probes/health/ResponsiveHealthIndicatorTest.java
@@ -9,8 +9,12 @@ package io.camunda.zeebe.gateway.impl.probes.health;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doNothing;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 import io.camunda.zeebe.client.CredentialsProvider;
@@ -70,7 +74,7 @@ public class ResponsiveHealthIndicatorTest {
         new ResponsiveHealthIndicator(TEST_CFG, ZEEBE_CLIENT_PROPERTIES_WITH_TEST_DURATION);
 
     final var spyHealthIndicator = spy(healthIndicator);
-    when(spyHealthIndicator.supplyZeebeClient()).thenReturn(null);
+    doNothing().when(spyHealthIndicator).initZeebeClient();
 
     // when
     final var actualHealth = spyHealthIndicator.health();
@@ -96,7 +100,7 @@ public class ResponsiveHealthIndicatorTest {
     final var healthIndicator =
         new ResponsiveHealthIndicator(TEST_CFG, ZEEBE_CLIENT_PROPERTIES_WITH_TEST_DURATION);
     final var spyHealthIndicator = spy(healthIndicator);
-    when(spyHealthIndicator.supplyZeebeClient()).thenReturn(mockZeebeClient);
+    when(spyHealthIndicator.getZeebeClient()).thenReturn(mockZeebeClient);
 
     // when
     final var actualHealth = spyHealthIndicator.health();
@@ -121,7 +125,13 @@ public class ResponsiveHealthIndicatorTest {
     final var healthIndicator =
         new ResponsiveHealthIndicator(TEST_CFG, ZEEBE_CLIENT_PROPERTIES_WITH_TEST_DURATION);
     final var spyHealthIndicator = spy(healthIndicator);
-    when(spyHealthIndicator.supplyZeebeClient()).thenReturn(mockZeebeClient);
+    doAnswer(
+            invocation -> {
+              spyHealthIndicator.setZeebeClient(mockZeebeClient);
+              return null;
+            })
+        .when(spyHealthIndicator)
+        .initZeebeClient();
 
     // when
     final var actualHealth = spyHealthIndicator.health();
@@ -129,6 +139,35 @@ public class ResponsiveHealthIndicatorTest {
     // then
     assertThat(actualHealth).isNotNull();
     assertThat(actualHealth.getStatus()).isSameAs(Status.UP);
+  }
+
+  @Test
+  public void shouldNotCloseZeebeClientInHealthChecks() {
+    // given
+    final var mockZeebeClient = mock(ZeebeClient.class);
+    final var mockTopologyRequestStep1 = mock(TopologyRequestStep1.class);
+    final var mockZeebeFuture = mock(ZeebeFuture.class);
+
+    // build mock chain
+    when(mockZeebeClient.newTopologyRequest()).thenReturn(mockTopologyRequestStep1);
+    when(mockTopologyRequestStep1.send()).thenReturn(mockZeebeFuture);
+
+    final var healthIndicator =
+        new ResponsiveHealthIndicator(TEST_CFG, ZEEBE_CLIENT_PROPERTIES_WITH_TEST_DURATION);
+    final var spyHealthIndicator = spy(healthIndicator);
+    doAnswer(
+            invocation -> {
+              spyHealthIndicator.setZeebeClient(mockZeebeClient);
+              return null;
+            })
+        .when(spyHealthIndicator)
+        .initZeebeClient();
+
+    // when
+    spyHealthIndicator.health();
+
+    // then
+    verify(mockZeebeClient, never()).close();
   }
 
   @Test
@@ -143,7 +182,7 @@ public class ResponsiveHealthIndicatorTest {
         new ResponsiveHealthIndicator(gatewayCfg, ZEEBE_CLIENT_PROPERTIES_WITH_TEST_DURATION);
 
     // when
-    final var actualZeebeClient = healthIndicator.supplyZeebeClient();
+    final var actualZeebeClient = healthIndicator.getZeebeClient();
 
     // then
     assertThat(actualZeebeClient).isNull();
@@ -160,9 +199,10 @@ public class ResponsiveHealthIndicatorTest {
 
     final var healthIndicator =
         new ResponsiveHealthIndicator(gatewayCfg, ZEEBE_CLIENT_PROPERTIES_WITH_TEST_DURATION);
+    healthIndicator.initZeebeClient();
 
     // when
-    final var actualZeebeClient = healthIndicator.supplyZeebeClient();
+    final var actualZeebeClient = healthIndicator.getZeebeClient();
 
     // then
     assertThat(actualZeebeClient).isNotNull();

